.github/workflows/build-checking.yml

```diff
@@ -0,0 +1,42 @@
+name: Check Build
+
+env:
+  DRAGONFLY_PATH: "${{github.workspace}}/build-opt/dragonfly"
+  
+on: 
+  schedule:
+    - cron: '0 7 */3 * *'
+  workflow_dispatch:
+      
+jobs:
+  build-n-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: true
```

Comment:
since we run periodically, it would help if you will also print the commit hash of the the git repo, in case someone needs to reproduce it.

Reply:
Okay. How do you want me to print it?

Reply:
just run `echo current commit is ... ` and whatever github environment variable that holds the commit sha. If no env variable exists, then extract it with git command that prints the current sha (git rev-parse HEAD)